Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/AggregateAdapter.cpp

```diff
@@ -112,11 +115,77 @@ namespace AZ::DocumentPropertyEditor
             {
                 m_parent->m_pathIndexToChildMaps.resize(adapterIndex + 1);
             }
+            else
+            {
+                // shift any subsequent entries by 1 to make room for this new entry, then add it
+                m_parent->ShiftChildIndices(adapterIndex, pathEntryIndex, 1);
+            }
             m_parent->m_pathIndexToChildMaps[adapterIndex][pathEntryIndex] = this;
         }
     }
 
-    size_t RowAggregateAdapter::AggregateNode::EntryCount()
+    bool RowAggregateAdapter::AggregateNode::RemoveEntry(size_t adapterIndex)
+    {
+        // first recurse to remove entries from the bottom up
+        if (m_pathIndexToChildMaps.size() > adapterIndex)
+        {
+            auto& childMap = m_pathIndexToChildMaps[adapterIndex];
+            while (!childMap.empty())
+            {
+                childMap.begin()->second->RemoveEntry(adapterIndex);
+            }
+        }
+
+        // find entry in parent's index map and remove it
+        auto& parentsMap = m_parent->m_pathIndexToChildMaps[adapterIndex];
```

Comment:
Should we check `adapterIndex` is in bounds with `m_parent->m_pathIndexToChildMaps` as well like we do above? Or is this something we can assume is OK if we get to this point?

Reply:
good call, fixed now :)